sky/skylet/subprocess_daemon.py

```diff
@@ -4,8 +4,10 @@
 """
```

Comment:
This script is widely used across the system, i.e. it is not only used in the log tailing, but also in the remote clusters, to make sure it cleans up the children processes correctly, especially for cancelling a pytorch program that could have processes on GPUs. We need to be careful with that.

One thing to check: does kill_pg kill grandchildren?